### PR TITLE
introduce PluginBase

### PR DESCRIPTION
**Pull request by rapus95**

### CODE REVIEW

File: src/HoJBot.jl [hunk 1]
```diff
@@ -22,6 +22,10 @@ import HTTP
 import JSON3
 import StructTypes
 
+include("pluginbase.jl")
+
+import .PluginBase: handle_command as handler, handle_observer as commander
```
Comment (rapus95) [inline src/HoJBot.jl:27]: this won't be necessary anymore once we moved every plugin over to the modular system.

**Comment by tk3369** [comment]
I guess you can get this in as part of #91. Close this one?

**Comment by rapus95** [comment]
@tk3369 you mean by merging unrelated topics into the same PR? 🙈 sure. The reason why it's also part of the other PR is because I rebased them on top of each other to have a streamlined merge process. But the can be pushed mostly independently nevertheless (by rebasing again) But i see where that comes from 🤔 Then I need a more generic PR name for the other one

**Comment by tk3369** [comment]
Yes, my perspective is that it is not easy to review a framework without an actual use case. So I'm fine with bundling it with the very first usage of that framework.

**Pull request closed without merging**